src/aiodynamo/models.py

```diff
@@ -203,10 +214,14 @@ def delays(self):
             )
             yield current_delay_secs
 
-    async def attempts(self) -> AsyncIterable[None]:
-        deadline = time.monotonic() + self.time_limit_secs
-        for delay in self.delays():
-            yield
-            if time.monotonic() > deadline:
-                raise Throttled()
-            await asyncio.sleep(delay)
+
+@dataclass(frozen=True)
+class ExponentialBackoffThrottling(ThrottleConfig):
+    base_delay_secs: int = 2
+    max_delay_secs: int = 20
+
+    def delays(self) -> Iterable[float]:
+        for attempt in count():
+            yield min(
+                random.random() * (self.base_delay_secs ** attempt), self.max_delay_secs
+            )
```

Comment:
I consider `Deco` already exponential...

Reply:
here too, I meant `DecorrelatedBackoffThrottling` is already exponential... I think 💭 🤔

Reply:
they behave different though. the Decorelated... I got from their blog post about throttling. this one is from botocore. naming isn't great, but let me run the tests with them to see how they work first.

Reply:
`ExponentialBackoffThrottling` passes the tests suite (with default args)

Reply:
`DecorelatedJitterThrottling` also passes.

Reply:
and just to make sure, disabling throttling makes the tests fail: ```python
class NoThrottle(ThrottleConfig):
    def delays(self):
        yield 0
        raise Exception('no throttling!')
```